homeassistant/helpers/template.py

```diff
@@ -372,7 +372,21 @@ def async_render(
         if self.hass.config.legacy_templates or not parse_result:
             return render_result
 
-        return self._parse_result(render_result)
+        parsed_result = self._parse_result(render_result)
+
+        # Objects serialized with `to_json` use double quotes in the string while
+        # objects serialized by jinja use single quotes. If we detect that to_json
+        # was used as a filter, there are double quotes in the string, and literal_eval
+        # returns a valid deserialized JSON type, we should return the rendered result
+        # as is
+        if (
```

Comment:
Instead of adding more of these, I think we should look into a more generic escape hatch. Allowing one always to force a string/raw result.

Reply:
I was just about to tag you when I realized this was related to https://github.com/home-assistant/core/pull/41227. I am open to suggestions on how to achieve this as I am just dipping my toes into the templating code and I'm already a bit lost